test/modules/http/compliance/http-custom-agent.test.ts

```diff
@@ -62,3 +62,12 @@ it('preserves the context of the "createConnection" function in a custom https a
   const [context] = createConnectionContextSpy.mock.calls[0]
   expect(context.constructor.name).toBe('CustomHttpsAgent')
 })
+
+it('forward the custom agent\'s options', async () => {
+  const request = https.request(httpServer.https.url('/resource'), {
+    agent: new https.Agent({ rejectUnauthorized: false }),
+  })
+  request.end()
+
+  await expect(waitForClientRequest(request)).resolves.not.toThrow()
```

Comment:
My only concern is that this test doesn't make it clear what's being asserted. Is the expectation for the request to complete without errors? Did it not do that before? We need to define the expectations here clearer, I think.

Reply:
I see your point. 
If the options are not propagated correctly, the request will fail due to its HTTPS nature, as we did not provide a certificate.

Reply:
I think I will remove this test. The option forwarding is implies in the `normalize*` function tests, which you've added.